### ISSUE TITLE
[Bug]: Raw text getting parsed as JSON

### ISSUE THREAD

**Issue by ClarksonCote** [appsmithorg/appsmith#11246]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I am trying to hit a POST endpoint to GraphQL. The GraphQL instance is designed to just accept a simplified query variation. This can be passed in Postman and the results are successful, however Appsmith appears to be trying to parse the text out as JSON, which is preventing a successful POST.

This seems to relate to the issue below that was solved, though possibly not for these specific circumstances?
#5561 

### Steps To Reproduce

Here's what I send in the Body:
`{ metadatarecord ( limit: 10 ) { assignedId  uploadTime } }`

It posts "successfully" with a syntax error being returned from the endpoint, in response to my attempt, and the body of the query Appsmith sent looks as follows due to erroneous parsing attempts and adding extraneous quotes, which I believe is causing the issue:
```
"body" : {
"metadatarecord ( limit":"10 ) { assignedId uploadTime"
}
```

If I instead attempt to store the desired body content as a string, and just pass in those contents:
`{{appsmith.store.queryString}}`

The resulting body text Appsmith sends is within quotations, which also prevents success...
`"body" : "{ metadatarecord ( limit: 10 ) { assignedId uploadTime } }"`

In short, there seems to be no way for me to just pass in the desired raw text for me to interface to the endpoint as designed. Just to repeat, there are no issues with this method through postman. Image of the approach there is also attached.

![postman](https://user-images.githubusercontent.com/99921274/154576186-541f1d8a-91b3-4ab0-be6f-1df66536770d.png)

### Environment

Production

### Version

Self-Hosted

**Comment by nidhi-nair** [comment]
Hey @ClarksonCote ! Quick question around the endpoint you're hitting. What content-type does it expect to work with? The standard GraphQL requests have a JSON, but this minified query seems like it could be plain text? 

Also, could you please check if your Appsmith query has any content-type pre-filled? Please share the value from both the `Headers` tab on the API pane, and the `Logs` tab in the debugger from your failed run.

**Comment by ClarksonCote** [comment]
@nidhi-nair yes, in the header, we use (and the GraphQL expects) content type to be `application/json`  Using any other type will result in an error, despite the body here being plain text.

Similar to the resolved issue I linked to though ( #5561 ), it seems that there would be a desire to allow plain text to be passed even when the content-type specification is `application/json`?

![graphql-headers](https://user-images.githubusercontent.com/99921274/154690920-b175a3bf-d92d-4dee-a6ed-a3af7e4c6782.png)
![graphql-headers-logs](https://user-images.githubusercontent.com/99921274/154690934-c30028c8-00bf-406b-8965-834575408973.png)

**Comment by nidhi-nair** [comment]
Ah I see. The issue and PR (#5561 and #5642) are actually a solve for exactly this issue, but for the responses returned from the APIs. Let me talk about this within the team and figure out what amount of validation we can get away with for this content-type.

**Comment by ClarksonCote** [comment]
@nidhi-nair thank you. As an attempted workaround on our end in the meantime, we changed our endpoint to accept `application/graphql` content-type posts as well, rather than just `application/json`. I updated Appsmith to that content-type in the header, and got further there, but have encountered another issue that will likely require me to post another bug submission...

When I put the query as raw text in the body of the post, it works fine. Great! Success!

However, when I store the same query in a variable and reference the variable as {{appsmith.store.variable}} in the body, I get a syntax error from my endpoint.

Here's the perplexing part... In both cases, the body of the request sent by Appsmith in the logs are IDENTICAL. There are no differences in quotes or anything else, yet in one case I get an error, and in the other case I don't.

**Comment by nidhi-nair** [comment]
Adding more details to the cause of this issue. For our REST API plugin, we're performing a validation step for all requests that have the header `Content-Type: application/json`. This validation loosely checks for a JSON representation of the body, and parses the body into this JSON as well.

For example, in this scenario, the input could be forced into a key-value pair by adding the quotes as below:

"body" : {
`"metadatarecord ( limit"`    :    `"10 ) { assignedId uploadTime"`
}


The solve for this issue would either be to remove the validation for the content type, perhaps replace it with stricter validation, or to take into account the fact that the user has entered the data in `RAW` mode and ignore validation in just that scenario. We'll need to think about whether there is a chance that we'd break existing well-configured JSON bodies with either fix. This could happen for JSON bodies that have ignored inserting quotes around their keys.

**Comment by AmanAgarwal041** [comment]
@ClarksonCote We have introduced Graphql datasource for creating such type of API which support gql. So closing this issue for the same reason.
